From c79d2db299305d569d4038e2275ac4619435d48b Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <quic_viswanat@quicinc.com>
Date: Mon, 11 Sep 2023 22:56:52 +0530
Subject: [PATCH] drivers: remoteproc: Export rproc_start and rproc_stop

rproc_start and rproc_stop APIs are used by q6v5_mpd driver
and by external client drivers (if required) to handle
recovery for multi-pd targets.

Change-Id: I51cac339074b5444f59d0539f23bbb14e4af7a08
Signed-off-by: Vignesh Viswanathan <quic_viswanat@quicinc.com>
---
 drivers/remoteproc/remoteproc_core.c | 6 ++++--
 include/linux/remoteproc.h           | 2 ++
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/remoteproc/remoteproc_core.c b/drivers/remoteproc/remoteproc_core.c
index c4d202655895..ddfed965ec2d 100644
--- a/drivers/remoteproc/remoteproc_core.c
+++ b/drivers/remoteproc/remoteproc_core.c
@@ -1272,7 +1272,7 @@ void rproc_resource_cleanup(struct rproc *rproc)
 }
 EXPORT_SYMBOL(rproc_resource_cleanup);
 
-static int rproc_start(struct rproc *rproc, const struct firmware *fw)
+int rproc_start(struct rproc *rproc, const struct firmware *fw)
 {
 	struct resource_table *loaded_table;
 	struct device *dev = &rproc->dev;
@@ -1336,6 +1336,7 @@ static int rproc_start(struct rproc *rproc, const struct firmware *fw)
 
 	return ret;
 }
+EXPORT_SYMBOL(rproc_start);
 
 static int __rproc_attach(struct rproc *rproc)
 {
@@ -1712,7 +1713,7 @@ static int rproc_trigger_auto_boot(struct rproc *rproc)
 	return ret;
 }
 
-static int rproc_stop(struct rproc *rproc, bool crashed)
+int rproc_stop(struct rproc *rproc, bool crashed)
 {
 	struct device *dev = &rproc->dev;
 	int ret;
@@ -1747,6 +1748,7 @@ static int rproc_stop(struct rproc *rproc, bool crashed)
 
 	return 0;
 }
+EXPORT_SYMBOL(rproc_stop);
 
 /*
  * __rproc_detach(): Does the opposite of __rproc_attach()
diff --git a/include/linux/remoteproc.h b/include/linux/remoteproc.h
index f3c0e0103e81..9e3cb8a7a09f 100644
--- a/include/linux/remoteproc.h
+++ b/include/linux/remoteproc.h
@@ -708,4 +708,6 @@ void rproc_add_subdev(struct rproc *rproc, struct rproc_subdev *subdev);
 
 void rproc_remove_subdev(struct rproc *rproc, struct rproc_subdev *subdev);
 
+int rproc_start(struct rproc *rproc, const struct firmware *fw);
+int rproc_stop(struct rproc *rproc, bool crashed);
 #endif /* REMOTEPROC_H */
-- 
2.34.1

